charmcraft.yaml

```diff
@@ -10,25 +32,75 @@ bases:
       - name: ubuntu
         channel: "22.04"
         architectures: [amd64]
-      - name: centos
-        channel: "7"
-        architectures: [amd64]
 
 parts:
   charm:
-    build-packages: [git]
-    charm-python-packages: [setuptools]
-
-  # Create a version file and pack it into the charm. This is dynamically generated
-  # as part of the build process for a charm to ensure that the git revision of the
-  # charm is always recorded in this version file.
-  version-file:
-    plugin: nil
     build-packages:
-      - git
+      - wget
     override-build: |
-      VERSION=$(git -C $CRAFT_PART_SRC/../../charm/src describe --dirty --always)
-      echo "Setting version to $VERSION"
-      echo $VERSION > $CRAFT_PART_INSTALL/version
-    stage:
-      - version
+       wget https://github.com/mej/nhc/releases/download/1.4.3/lbnl-nhc-1.4.3.tar.gz
+       craftctl default
+
+requires:
+  fluentbit:
+    interface: fluentbit
+
+provides:
+  slurmctld:
+    interface: slurmd
+    limit: 1
+
+config:
+  options:
+    partition-config:
+      type: string
+      default: ""
+      description: >
+        Additional partition configuration parameters, specified as a space separated `key=value`
+        in a single line. Find a list of all possible partition configuration parameters
+        [here](https://slurm.schedmd.com/slurm.conf.html#SECTION_PARTITION-CONFIGURATION).
+        
+  
+        Example usage:
+        ```bash
+         $ juju config slurmd partition-config="DefaultTime=45:00 MaxTime=1:00:00"
+        ```
+
+    nhc-conf:
+      default: ""
+      type: string
+      description: >
+        Multiline string.
+        These lines are appended to the `nhc.conf` maintained by the charm.
+
+        Example usage:
+        ```bash
+         $ juju config slurmd nhc-conf="$(cat extra-nhc.conf)"
+        ```
+
+actions:
+  node-configured:
```

Comment:
Potentially better to have as `set-node-state` and then an enum with all the possible Slurm node states? ACTIVE, DRAINED, DOWN, etc.?

Reply:
Totally.... The reason we didn't do the way that you describe initially, is because we didn't want to track and/or modify the node state with slurm configuration, we simply wanted a way to take it out of `DownNodes` and rescue the node. Possibly we could track the node state though ... then your solution might be a reasonable replacement.

Reply:
Makes sense especially after you explained your reasoning behind it; you want an easy way to control whether a node is ready to have jobs scheduled on it. Perhaps rather than having an action like `set-node-state` where you invariably introduce more edge cases you need to handle, you could have something like `mark-active` and `mark-down` to function as a boolean switch for taking nodes offline for maintenance.